### PR TITLE
Fix Sphinx warnings

### PR DESCRIPTION
**Pull request by zariiii9003**
I fixed all Sphinx warnings except for the nixnet interface, that one remains broken.
The documentation is now built in github actions.

### CODE REVIEW

**Comment by zariiii9003** [comment]
@hardbyte Can you check whether the readthedocs build works as configured in `.readthedocs.yml`? Or could you somehow add me to the rtd admins?
Also: What do you think about the `furo` theme? It's the one that [setuptools](https://setuptools.pypa.io/en/latest/references/keywords.html) uses.

**Comment by hardbyte** [comment]
@zariiii9003 the `furo` theme looks great. I've invited you as an admin to python-can on readthedocs.

File: .readthedocs.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# .readthedocs.yaml
+# Read the Docs configuration file
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
```
Comment (hardbyte) [inline .readthedocs.yml:3]: 👍🏼

**Review by hardbyte — APPROVED** [review]

**Comment by zariiii9003** [comment]
> https://docs.readthedocs.io/en/stable/config-file/v2.html

Thank you!

I will create another PR for furo.

**Comment by zariiii9003** [comment]
@hardbyte Can you resync the rtd webhook? You can find it in Admin -> Inegrations -> GitHub incoming webhook -> Resync webhook. New pull requests should trigger a rtd rebuild once it is synced.

I do not have the permissions to to do that.

**Comment by hardbyte** [comment]
Hmm resync didn't seem to work so I've deleted the webhook and added the read the docs integration from scratch. Thanks!

**Pull request merged**